core/types/transaction.go

```diff
@@ -648,7 +649,7 @@ func NewMessage(from common.Address, to *common.Address, nonce uint64, amount *b
 }
 
 // AsMessage returns the transaction as a core.Message.
-func (tx *Transaction) AsMessage(s Signer, baseFee *big.Int) (Message, error) {
+func (tx *Transaction) AsMessage(s Signer, baseFee, l1Cost *big.Int) (Message, error) {
```

Comment:
I think we should make the `l1Cost` optional, to avoid changing all the callers of `AsMessage`.